bee-api/bee-rest-api/src/endpoints/config.rs

```diff
@@ -152,24 +134,38 @@ impl RestApiConfigBuilder {
         let port = multi_addr
             .iter()
             .find_map(|x| if let Protocol::Tcp(port) = x { Some(port) } else { None })
-            .unwrap_or_else(|| panic!("Unsupported protocol"));
-        let public_routes: Box<[String]> = self
-            .public_routes
-            .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(|s| s.to_string()).collect())
-            .into_boxed_slice();
-        let allowed_ips: Box<[IpAddr]> = self
-            .allowed_ips
-            .unwrap_or_else(|| DEFAULT_ALLOWED_IPS.to_vec())
-            .into_boxed_slice();
+            .expect("unsupported protocol");
+
+        let jwt_salt = self.jwt_salt.unwrap_or_else(|| DEFAULT_JWT_SALT.to_string());
+
+        let public_routes = {
+            let routes = self
+                .public_routes
+                .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(ToString::to_string).collect());
+            RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<_>>())
```

Comment:
I think you can do `map(route_to_regex)` directly

Reply:
Hmm also @Alex6323 mentioned this, but doesn't really work
Seems like it's necessary (and maybe thats the reason clippy isn't complaining).
Changing here the closures requires changes to the route_to_regex or requires additional conversions.

`routes.iter().map(route_to_regex)` would require `route_to_regex(&String)`, but it's defined as `route_to_regex(&str)`.
Or do you have any other thought how to change this?

Reply:
you could change `route_to_regex` argument to receive `impl core::ops::Deref<Target = str>` instead of `&str`. But I think I would leave this unmodified instead.